### DISCUSSION TITLE
SFTPGO and Power Automate (Microsoft Power Platform)

### DISCUSSION THREAD

**Discussion by cord-agencyroot** [drakkan/sftpgo#1355] [Q&A]
I am trying to connect Microsoft Power Platform (SFTP - SSH) connection to my hosted SFTPGo instance, but getting an error message:   `Bad Gateway: "server response does not contain SSH protocol identification error". `

I can connect fine using FileZilla or WinSCP. I am just using a Username/Password and not using a key for authentication and since this is hosted inside a Kubernetes cluster, we are using port a custom port 2022 to forward it to the SFTPGo container.

Has anyone else used Power Automate with SFTPGo and run into a similar issue?

**Comment by rel33** [comment]
There is something written about throttling: https://learn.microsoft.com/en-us/answers/questions/855221/error-code22665-detailsfailed-to-connect-to-sftp-s 

If you not plan to upload files, but only automate some administration tasks, there is REST API:
https://github.com/drakkan/sftpgo/blob/main/docs/rest-api.md

**Reply by cord-agencyroot** [reply]
Yeah, we have no throttling setup so it 'should' be unlimited. We decided to go in a different route but just found it odd that we ran into this issue with Microsoft's solution, but any other SFTP client we used connected fine. We are not sure if its a setup issue with our kubernetes cluster or if something on the SFTP GO side or Microsofts. Using the API is a good suggestion but this was for transfering files so not sure if that would work in this case.

**Comment by alexanderreichel** [comment]
Is there a known working solution already? I'm trying to create a new connection in Power Automate using the "Connect to SFTP - SSH" template. I am getting BadGateway error all the way while WinSCP, FileZilla, terminal etc. all work fine.
